browser_history/generic.py

```diff
@@ -377,3 +377,23 @@ def default(self, o):
             )
 
         return json_string
+
+    def save(self, filename, output_format="infer"):
+        """
+        Saves history or bookmarks to a file. Infers the type from the given filename
+        extension. If the type could not be inferred, it defaults to csv.
+
+        :param filename: the name of the file.
+        :param output_format: (optional)One the formats in `csv`, `json`, `jsonl`.
+            If not given, it will automatically be inferd from the file's extension
+        """
+        if output_format == "infer":
+            output_format = os.path.splitext(filename)[1][1:]
+            if  not output_format in self.format_map:
+                raise ValueError(
+                    f"Invalid extension .{output_format}. Should be one of "
+                    f"{self.format_map.keys()}"
```

Comment:
This shows up as `ValueError: Invalid extension .haha. Should be one of dict_keys(['csv', 'json', 'jsonl'])`

You can format it into a string to make it look a little better.
```suggestion
                    f"{', '.join(self.format_map.keys())}"
```

Reply:
Done, I had done it that way because it was how it was done elsewhere in the code.